Bearded.UI/Controls/implementations/ListControl.cs

```diff
@@ -327,35 +342,41 @@ private void addCellsDownwards()
             }
         }
 
-        private (Control Control, int Index, double Offset, double Height)
+        private (Control? Control, int Index, double Offset, double Height)
             addCellBelow(int index, double top)
         {
+            if (itemSource == null)
+                throw new InvalidOperationException("The ItemSource is not set.");
+
             var height = itemSource.HeightOfItemAt(index);
             var bottom = top + height;
 
             var cell = createCellIfVisible(index, bottom, top, height);
 
             if (cell.Control != null)
-                cells.AddLast(cell);
+                cells.AddLast((cell.Control, cell.Index, cell.Offset, cell.Height));
 
             return cell;
         }
 
-        private (Control Control, int Index, double Offset, double Height)
+        private (Control? Control, int Index, double Offset, double Height)
             addCellAbove(int index, double bottom)
         {
+            if (itemSource == null)
+                throw new InvalidOperationException("The ItemSource is not set.");
+
             var height = itemSource.HeightOfItemAt(index);
             var top = bottom - height;
 
             var cell = createCellIfVisible(index, bottom, top, height);
 
             if (cell.Control != null)
-                cells.AddFirst(cell);
+                cells.AddFirst((cell.Control, cell.Index, cell.Offset, cell.Height));
```

Comment:
This looks like we should be able to write it nicer instead of having to recreate almost the same struct.

Reply:
Do you know how? These structs are different (the first entry has a different type - nullable and non nullable).
I really don't want to change the type of `cells` to always assume the control can be null.

I'm assuming here that the compiler optimizes this to the same code, but I didn't check that.

Reply:
I was hoping we could make the whole tuple nullable but looks like we need the associated data in either case. It's a little annoying, but I don't see an easier way of improving it. It could probably be refactored, but perhaps that's for another time.